src/main/java/com/coveo/spillway/storage/RedisStorage.java

```diff
@@ -152,11 +148,11 @@ public Map<LimitKey, Integer> addAndGetWithLimit(Collection<AddAndGetRequest> re
                       Collections.singletonList(redisKey),
                       Arrays.asList(
                           String.valueOf(request.getCost()), String.valueOf(request.getLimit()))));
-              pipeline.expire(redisKey, (int) request.getExpiration().getSeconds() * 2);
-              pipeline.exec();
+              pipeline.expire(redisKey, request.getExpiration().getSeconds() * 2);
             });
+
         pipeline.sync();
-      } catch (IOException e) {
+      } catch (Throwable e) {
         logger.error("Unable to close redis storage pipeline.", e);
```

Comment:
same here

Reply:
I will keep it now I think because it makes sense again.

Reply:
if we're catching Throwable, it's not only related to closing the pipeline anymore